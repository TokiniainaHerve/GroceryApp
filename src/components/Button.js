import { useTheme } from '@react-navigation/native';
import PropTypes from 'prop-types';
import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import { labelTypes, typography,shadow } from '@/theme';
import LinearGradient from 'react-native-linear-gradient';
import { TextLabel } from './TextLabel';
const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    
  },
  linear_gradient:{
    width:'100%',
    height:60,
    alignItems:'center',
    justifyContent:'center',
    borderRadius:8
  }
});

export function Button({ style, textStyle, title, ...rest }) {
  const { colors } = useTheme();

  return (
    <TouchableOpacity style={[styles.button, { borderColor: colors.border },shadow.primary, style]} {...rest}>
      <LinearGradient start={{x: 0, y: 0}} end={{x: 1, y: 0}} colors={[colors.primary, colors.primary_dark]} style={styles.linear_gradient}>
        <TextLabel text={title} type={labelTypes.PARAGRAPH_REGULAR} style={{color:colors.base}} />
      </LinearGradient>
    </TouchableOpacity>
  );
}

Button.propTypes = {
  style: PropTypes.object,
  textStyle: PropTypes.object,
  title: PropTypes.string.isRequired,
};

Button.defaultProps = {
  style: null,
  textStyle: null,
};
