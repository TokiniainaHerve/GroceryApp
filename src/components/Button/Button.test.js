import {  render} from '@testing-library/react-native';
import React from 'react';
import {Button} from './Button'
describe('Button', () => {
    it('should match the snapshot', () => {
      const { toJSON } = render(<Button title={"test button"}/>);
      expect(toJSON()).toMatchSnapshot();
    });
  
 
});