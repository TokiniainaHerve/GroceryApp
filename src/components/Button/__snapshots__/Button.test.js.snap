// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "rgb(216, 216, 216)",
      "elevation": 5,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
      "width": "100%",
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4278221567,
        undefined,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "height": 60,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins-Regular",
            "fontSize": 16,
          },
          Object {
            "color": "rgb(28, 28, 30)",
          },
          Object {
            "color": undefined,
          },
        ]
      }
    >
      test button
    </Text>
  </BVLinearGradient>
</View>
`;
